### ISSUE TITLE
ADD a warning notification when all missing mod is not present before proceeding to install

### ISSUE THREAD

**Issue by ttax00** [PrismLauncher/PrismLauncher#822]
### Operating System

Linux
Fedora 37

### Version of Prism Launcher

Prism Launcher 6.3
Installed through flatpack

### Version of Qt

Qt 5.15.8

### Description of bug

AOF6 modpack installed with missing mods, yet no errors were thrown.

Apparently, these are the only mods that were missing out of +300
Prism log:
```console
   18.213 C Exception: "Invalid URL: "
   18.213 D Blocked mod on curseforge "entityculling-fabric-1.5.2-mc1.19.jar"
   18.213 C Exception: "Invalid URL: "
   18.213 D Blocked mod on curseforge "fabric-experiencebugfix-1.19-18.jar"
   18.214 C Exception: "Invalid URL: "
   18.214 D Blocked mod on curseforge "ToastManager-1.19.2-1.0.1.jar"
   18.215 C Exception: "Invalid URL: "
   18.215 D Blocked mod on curseforge "Structory-1.19-v1.3.jar"
   18.215 C Exception: "Invalid URL: "
   18.215 D Blocked mod on curseforge "Structory-1.19-v1.3.jar"
   18.215 C Exception: "Invalid URL: "
   18.215 D Blocked mod on curseforge "convenientnametags-0.3.0.jar"
   18.215 C Exception: "Invalid URL: "
   18.215 D Blocked mod on curseforge "wolf-armor-compat-1.0.2-1.19.2.jar"
   18.215 C Exception: "Invalid URL: "
   18.216 D Blocked mod on curseforge "takesapillage-1.19-1.0.4.jar"
   18.216 C Exception: "Invalid URL: "
   18.216 D Blocked mod on curseforge "ctov-3.1.4.jar"
   18.216 C Exception: "Invalid URL: "
   18.216 D Blocked mod on curseforge "muffins-slimegolem-1.19.x-1.0.2-fabric.jar"
   18.216 C Exception: "Invalid URL: "
   18.216 D Blocked mod on curseforge "deeperdarker-fabric-1.1.6-fabric.jar"
   18.216 C Exception: "Invalid URL: "
   18.216 D Blocked mod on curseforge "deeperdarker-fabric-1.1.6-fabric.jar"
   18.216 C Exception: "Invalid URL: "
   18.216 D Blocked mod on curseforge "fish_of_thieves-mc1.19.2-v2.1.1-fabric.jar"
   18.217 C Exception: "Invalid URL: "
   18.217 D Blocked mod on curseforge "minecells-1.3.3.jar"
   18.217 C Exception: "Invalid URL: "
   18.217 D Blocked mod on curseforge "Structory_Towers_1.19_v1.0.1.jar"
   18.217 C Exception: "Invalid URL: "
   18.217 D Blocked mod on curseforge "netherdepthsupgrade-fabric-2.0.2-1.19.2.jar"
   18.217 C Exception: "Invalid URL: "
   18.217 D Blocked mod on curseforge "convenientdecor-0.0.3.jar"
```

### Steps to reproduce

Add instance > Search AOF > AOF6 > install All of fabric 1.3.0

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by flowln** [comment]
did this dialog not show up for you?
![blocked mods from CurseForge dialog, prompting the user to manually download the missing mods](https://user-images.githubusercontent.com/9145768/216769434-c53e23aa-31a4-4cf8-bb24-b7f300450452.png)

**Comment by ttax00** [comment]
Ooh, It did! But the size of the dialog box made me think it was only missing `expereiencebugfix`. And Ok did not warn of the remaining. 

Might be a good think to double check the users if some are still not found. 
It wasn't a bug then. thanks @flowln

**Comment by Frontear** [comment]
i think we really should disable the Okay button until all mods are detected.

**Comment by ttax00** [comment]
> i think we really should disable the Okay button until all mods are detected.

Should we close this issue, and make a feature request instead?

**Comment by flowln** [comment]
> i think we really should disable the Okay button until all mods are detected.

I disagree, often times I want to download modpacks for testing some stuff, and not having to download the missing mods speeds up the process a lot. I think just having a confirmation or more clearly telling the user that something is missing is enough.

**Comment by Xetaxheb** [comment]
A simple "You have not downloaded all of the missing mods, are you sure you want to continue? Yes/No" would fix this.

**Comment by Frontear** [comment]
> > i think we really should disable the Okay button until all mods are detected.
> 
> I disagree, often times I want to download modpacks for testing some stuff, and not having to download the missing mods speeds up the process a lot. I think just having a confirmation or more clearly telling the user that something is missing is enough.

i feel like this use-case doesn't make a lot of sense because if youre missing mods that are intended to be in a pack, you can have a range of errors from missing content to straight up not being able to launch because of missing dependencies. There's not a lot to test in terms of a modpack if there's a possibility its not even gonna work right in the first place. And really, how much more time is someone saving not downloading the 5-10 mods that were blocked compared to the 200~ that weren't?

**Comment by flowln** [comment]
> i feel like this use-case doesn't make a lot of sense because if youre missing mods that are intended to be in a pack, you can have a range of errors from missing content to straight up not being able to launch because of missing dependencies. There's not a lot to test in terms of a modpack if there's a possibility its not even gonna work right in the first place. And really, how much more time is someone saving not downloading the 5-10 mods that were blocked compared to the 200~ that weren't?

And how is it not solved by having just a warning?

I work a lot on modpack download, so I often have to download modpacks just to check if they install correctly, not to actually play the game. For sure, it's a small use-case, but with a warning I can still have that, while the average user will probably never see the warning, and when they do they can take a step back and download the missing things.

Thus, I don't see how disabling the option to skip that step is better than just making it harder for the uninitiated.

**Comment by ttax00** [comment]
I think we should relable this thread as `enhancement` instead, since we're talking about a warning pop up feature.
